scrapping.js

```diff
@@ -0,0 +1,70 @@
+
+/* import */
+const axios = require('axios')
+const cheerio = require('cheerio')
+const url = "https://news.daum.net/"
+
+async function getNews() {
+
+    let resource = {
+
+        data: [],
+        message: ""
+    }
+
+    try {
+
+        const response = await axios.get(url)
+        const $ = cheerio.load(response.data)
+
+        let extractData = $('ul.list_newsissue')
+                    .find('ul li')
+                    .map((i, el) => {
+
+                        const title = extractTitle($, el)
+                        const image = extractImage($, el)
+
+                    return {title, image}
+                    }).get()
+
+        resource.data = extractData
+        resource.message = "Successful extraction of news data "
+
+        return resource
```

Comment:
함수가 분리된 건 확인 했는데, 리펙터링은 유지보수와 가독성을 위한 장치기 때문에 더 효과적으로 분리가 되면 좋겠어요.
그리고 Title을 추출한다, Image를 추출한다는 맥락에는 아래 구문까지 포함이 된 내용이라고 생각이 되요
```
        let extractData = $('ul.list_newsissue')
                    .find('ul li')
                    .map((i, el) => {
```

저라면 각 함수의 로직 안에 find와 map에 해당하는 로직도 포함시키고, 각각 오브젝트를 반환하도록 할거 같아요.

```suggestion
    try {
        const response = await axios.get(url)
        const $ = cheerio.load(response.data)

        const titles = extractTitles($)
        const images = extractImages($)

        return {
                data: {titles, images},
                message: "Successful extraction of news data",
         }
```

Reply:
let을 활용하는 문법은 장기적으로 유지보수와 컨텍스트 파악이 어렵게 하는 주된 요인이 되요.
그래서 저는 let사용을 가급적 지양하는 편이에요.
대부분의 경우에서 let을 활용하지 않고도 로직을 해결할 수 있기 때문에 충분한 고민을 해서 로직을 작성해 보면 더 좋은 코드가 나올거에요.

Reply:
https://github.com/DAALZZWI/NewsApi/pull/3#discussion_r1155070994

제시해주신 코드 보니까 변수 선언을 좀 줄이고 가독성 있게 수정해야 할 꺼 같습니다
수정 되 는대로 바로 올리겠습니다

Reply:
>let을 활용하는 문법은 장기적으로 유지보수와 컨텍스트 파악이 어렵게 하는 주된 요인이 되요.
그래서 저는 let사용을 가급적 지양하는 편이에요.
대부분의 경우에서 let을 활용하지 않고도 로직을 해결할 수 있기 때문에 충분한 고민을 해서 로직을 작성해 보면 더 좋은 코드가 나올거에요.

var의 중복변수 선언 가능 때문에 let을 썻었는데 let도 생각해보니 남발해서 쓰면
소스코드가 많아질때 많은 불편함이 있을꺼 같습니다 
되도록이면 const 쓰겠습니다

Reply:
>let을 활용하는 문법은 장기적으로 유지보수와 컨텍스트 파악이 어렵게 하는 주된 요인이 되요.
그래서 저는 let사용을 가급적 지양하는 편이에요.
대부분의 경우에서 let을 활용하지 않고도 로직을 해결할 수 있기 때문에 충분한 고민을 해서 로직을 작성해 보면 더 좋은 코드가 나올거에요.

https://github.com/DAALZZWI/NewsApi/pull/3/commits/2ff25c489718e7ce92085b7a69c271954d615b77

resource로 썻던 let 객체는 삭제하고 return 으로 바로 반환되도록 코드 구성했습니다
검토 부탁 드립니다

Reply:
https://github.com/DAALZZWI/NewsApi/pull/3/commits/cfb27e733597b32f45a44d864e49c92a4c1f4b8c

titles과 images를 분리를 했으나 titles 따로 images 따로 뭉쳐서 데이터가 나오는 현상이 있었습니다
배열의 인덱스에 맞춰 titles 하나 images 하나 짝지어서 데이터를 재가공 시키는 manufacture함수를 
추가 시켰습니다
검토 부탁 드립니다

Reply:
리뷰할 때 데이터가 깨지는건 고려 못했네요.
로직이 조금 지저분해 보여서 제안드렸던 건데, 해당 리뷰에 맞추기 위해 굳이 maufacture와 같은 함수를 억지로 추가하는건 효율적이지 못한거 같아요.

코드 제안드렸던 건 무시하시고, 불필요한 변수 선언을 제거하고  방식으로 다시 리펙토링 해보면 좋을 것 같아요.

Reply:
https://github.com/DAALZZWI/NewsApi/pull/3/commits/76ddca47785b8d29645ecf1fa4d097d2e3bf4900

기존 방식으로 다시 돌아갔습니다
제가 생각했을 땐 maufacture 함수를 제거 한다고 가정 했을땐 이전 코드가 가독성이 있다고 
생각하여 리턴 시켰습니다 

그리고 결과적으로 scrapping.js에서 간혹가다가 스크래핑을 수행 못할 때가 있습니다
그럴때 try catch를 안 할시 서버가 꺼져버리는 현상으로 부득이하게 넣었습니다
검토 부탁 드립니다